src/main/kotlin/com/exactpro/th2/codec/api/impl/ThreadSafeCodec.kt

```diff
@@ -48,13 +52,15 @@ class ThreadSafeCodec(
 
     private fun getInstance() = instances.computeIfAbsent(Thread.currentThread()) {
         codecFactory().apply {
-            init(dictionary, settings)
+            dictionary.inputStream().use { stream ->
+                init(stream, settings)
```

Comment:
Read dictionary for every thread is not optimal. We need a function to load a dictionary before init